test/unittests/test_parse_en.py

```diff
@@ -817,6 +861,463 @@ def test_extract_date_with_may_I_en(self):
+        self.assertEqual(
+            extract_datetime('i have things to do in 1 weekend', dt)[0],
+            datetime(2017, 6, 5, tzinfo=default_timezone()))
+        self.assertEqual(
+            extract_datetime('i have things to do in 2 weekends', dt)[0],
+            datetime(2017, 6, 12, tzinfo=default_timezone()))
+
+        self.assertEqual(
+            extract_datetime('i had things to do last weekend', dt)[0],
+            extract_datetime('i had things to do last saturday', dt)[0])
+        self.assertEqual(
+            extract_datetime('i had things to do last weekend', dt)[0],
+            datetime(2017, 5, 27, tzinfo=default_timezone()))
+        self.assertEqual(
+            extract_datetime('i had things to do past weekend', dt)[0],
+            datetime(2017, 5, 27, tzinfo=default_timezone()))
+
+        self.assertEqual(
+            extract_datetime('i had things to do a weekend ago', dt)[0],
+            extract_datetime('i had things to do last friday', dt)[0])
+        self.assertEqual(
+            extract_datetime('i had things to do a weekend ago', dt)[0],
+            datetime(2017, 5, 26, tzinfo=default_timezone()))
+        self.assertEqual(
+            extract_datetime('i had things to do 2 weekends ago', dt)[0],
+            datetime(2017, 5, 19, tzinfo=default_timezone()))
+
+    def test_extract_next_en(self):
+        dt = datetime(2017, 6, 1, 0, 0, 0)
+        # next {timedelta unit}
+        self.assertEqual(
+            extract_datetime('i have things to do next second', dt)[0],
+            datetime(2017, 6, 1, 0, 0, 1, tzinfo=default_timezone()))
+        self.assertEqual(
+            extract_datetime('i have things to do next minute', dt)[0],
+            datetime(2017, 6, 1, 0, 1, 0, tzinfo=default_timezone()))
+        self.assertEqual(
+            extract_datetime('i have things to do next hour', dt)[0],
+            datetime(2017, 6, 1, 1, 0, 0, tzinfo=default_timezone()))
+        self.assertEqual(
+            extract_datetime('i have things to do next day', dt)[0],
+            datetime(2017, 6, 2, tzinfo=default_timezone()))
+
+        # 1st day of {calendar unit}
+        self.assertEqual(
+            extract_datetime('i have things to do next week', dt)[0],
+            extract_datetime('i have things to do next monday', dt)[0])
+        self.assertEqual(
+            extract_datetime('i have things to do next week', dt)[0],
+            datetime(2017, 6, 5, tzinfo=default_timezone()))
+        self.assertEqual(
+            extract_datetime('i have things to do next month', dt)[0],
+            datetime(2017, 7, 1, tzinfo=default_timezone()))
+        self.assertEqual(
+            extract_datetime('i have things to do next year', dt)[0],
+            datetime(2018, 1, 1, tzinfo=default_timezone()))
+
+        # old test moved here due to new disambiguation between "next week" and "in a week"
+        # testExtract("remind me to call mom next week",
```

Comment:
Comment here is implemented below; consider removing comment?

Reply:
can be removed, is just documenting the difference in expected output, this test expects different things in mycroft and ovos

Reply:
I see, missed that the times are different